__tests__/app.test.js

```diff
@@ -1,227 +1,258 @@
+    const comment = {
+      username: "butter_bridge",
+      body: "I am commenting for Eric!",
+    };
+    return request(app)
+      .post("/api/articles/nonExistenArticleId/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("POST:400 sends an appropriate status and error message when given an invalid comment", () => {
+    const comment = {};
+    return request(app)
+      .post("/api/articles/1/comments")
       .send(comment)
       .expect(400)
-      .then(({body}) => {
-          expect(body.msg).toBe('Bad Request')
-      })
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
   });
 
-  test('POST:404 sends an appropriate status and error message when given a valid a non-existent article_id', () => {
-      const comment = {
-          username: 'butter_bridge',
-          body: 'I am commenting for Eric!'
-      }
-
-      return request(app)
-        .post('/api/articles/999/comments')
-        .send(comment)
-        .expect(404)
-        .then(({body}) => {
-          expect(body.msg).toBe('article does not exist');
-        });
-    });
+  test("POST:404 sends an appropriate status and error message when given a valid a non-existent article_id", () => {
+    const comment = {
+      username: "butter_bridge",
+      body: "I am commenting for Eric!",
+    };
+
+    return request(app)
+      .post("/api/articles/999/comments")
+      .send(comment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("article does not exist");
+      });
+  });
 });
 
+describe("PATCH /api/articles/:article_id", () => {
+  test("should update an article by article_id", () => {
+  const existingArticle = {
+    author: "butter_bridge",
+    title: "Living in the shadow of a great man",
+    article_id: 1,
+    body: "I find this existence challenging",
+    topic: "mitch",
+    created_at: 1594329060000,
+    votes: 100,
+    article_img_url:
+      "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+  };
+
+    const newVote = { inc_votes: 8 };
+
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newVote)
+      .expect(200)
+      .then(({ body }) => {
+        const updatedArticle = body;
+        expect(updatedArticle).toMatchObject({
+          author: "butter_bridge",
+          title: "Living in the shadow of a great man",
+          article_id: 1,
+          body: "I find this existence challenging",
+          topic: "mitch",
+          created_at: expect.any(String),
+          votes: existingArticle.votes + newVote.inc_votes,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
```

Comment:
You will need to consider error handling tests for your patch requests before you can merge to main

Reply:
yes..my bad
just did and doing another nchelp pr

Reply:
done now